src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -353,13 +353,14 @@ private String getRepoUrl(ScanRequest request, File cxFile) {
     }
 
     private void sendSubmittedScanEmail(ScanRequest request) {
-        Map<String, Object> emailCtx = new HashMap<>();
-
-        emailCtx.put("message", "Checkmarx Scan has been submitted for "
-                .concat(request.getNamespace()).concat("/").concat(request.getRepoName()).concat(" - ")
-                .concat(request.getRepoUrl()));
-        emailCtx.put("heading", "Scan Request Submitted");
-        emailService.sendmail(request.getEmail(), "Checkmarx Scan Submitted for ".concat(request.getNamespace()).concat("/").concat(request.getRepoName()), emailCtx, "message.html");
+        if (flowProperties.getMail() != null && flowProperties.getMail().isNotificationEnabled()) {
+            Map<String, Object> emailCtx = new HashMap<>();
+            emailCtx.put("message", "Checkmarx Scan has been submitted for "
+                    .concat(request.getNamespace()).concat("/").concat(request.getRepoName()).concat(" - ")
+                    .concat(request.getRepoUrl()));
+            emailCtx.put("heading", "Scan Request Submitted");
+            emailService.sendmail(request.getEmail(), "Checkmarx Scan Submitted for ".concat(request.getNamespace()).concat("/").concat(request.getRepoName()), emailCtx, "message.html");
```

Comment:
it's better to move the structure of the email to the email service

Reply:
Done! All email related logic is now in one class.